rotkehlchen/tests/unit/test_chains_aggregator.py

```diff
@@ -26,42 +27,81 @@ def test_module_deactivation(blockchain):
         assert module_name not in blockchain.eth_modules
 
 
-def test_filter_active_evm_addresses(blockchain):
+@pytest.mark.parametrize('ethereum_accounts', [[]])
+def test_detect_evm_accounts(blockchain: 'ChainsAggregator'):
+    eth_addy_contract = make_evm_address()
+    addy_eoa = make_evm_address()
+    initial_accounts_data = [
+        BlockchainAccountData(
+            chain=SupportedBlockchain.ETHEREUM,
+            address=eth_addy_contract,
+        ),
+        BlockchainAccountData(
+            chain=SupportedBlockchain.OPTIMISM,
+            address=addy_eoa,
+        ),
+    ]
 
-    contract_addy = make_evm_address()
-    all_addy = make_evm_address()
-    optimism_addy = make_evm_address()
-    avalanche_addy = make_evm_address()
+    blockchain.modify_blockchain_accounts(
+        blockchain=SupportedBlockchain.ETHEREUM,
+        accounts=[eth_addy_contract],
+        append_or_remove='append',
```

Comment:
can you also test and address in ethereum that is not a contract? since ethereum is a case that has different paths in the logic